src/connector/src/parser/avro/glue_resolver.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use apache_avro::Schema;
+use aws_sdk_glue::types::{SchemaId, SchemaVersionNumber};
+use aws_sdk_glue::Client;
+use moka::future::Cache;
+
+use crate::connector_common::AwsAuthProps;
+use crate::error::ConnectorResult;
+
+/// Fetch schemas from AWS Glue schema registry and cache them.
+///
+/// Background: This is mainly used for Avro **writer schema** (during schema evolution): When decoding an Avro message,
+/// we must get the message's schema id, and use the *exactly same schema* to decode the message, and then
+/// convert it with the reader schema. (This is also why Avro has to be used with a schema registry instead of a static schema file.)
+///
+/// TODO: support protobuf (not sure if it's needed)
+pub trait GlueSchemaCache {
+    /// Gets the a specific schema by id, which is used as *writer schema*.
+    async fn get_by_id(&self, schema_version_id: uuid::Uuid) -> ConnectorResult<Arc<Schema>>;
+    /// Gets the latest schema by arn, which is used as *reader schema*.
+    async fn get_by_name(&self, schema_arn: &str) -> ConnectorResult<Arc<Schema>>;
+}
+
+#[derive(Debug)]
+pub enum GlueSchemaCacheImpl {
+    Real(RealGlueSchemaCache),
+    Mock(MockGlueSchemaCache),
+}
```

Comment:
This isn't a good style to me: the trait seems a little unnecessary to me, but anyway

Reply:
Unnecessary as a tool (generic) but it is a doc to make sure `RealGlueSchemaCache` and `MockGlueSchemaCache` have the same interface.

The unnecessary part is the enum, as we can simply `Box<dyn GlueSchemaCache>`. This aligns more with mocked interfaces in Java or Go, but in rust we always prefer static dispatch with enum rather than dynamic dispatch (?)